Added new functions to register users and reset password

tiagomgelias opened the pull request.

tiagomgelias committed:
Added new functions to register users and reset password

tiagomgelias committed:
changes in function documentation

tiagomgelias committed:
Changes on setActive function parameters and documentation

claudio-silva commented on Interfaces/UserInterface.php:
```diff
@@ -168,4 +194,49 @@ function usernameField ($set = null);
+   * @param string $token
+   * @param int $id
+   */
+  function updateRememberToken ($token, $id);
```
Porque é que é necessário um $id aqui? O próprio objecto sabe internamente qual é o seu ID, logo a classe que o implementa não precisa deste argumento.

claudio-silva commented on Interfaces/UserInterface.php:
```diff
@@ -168,4 +194,49 @@ function usernameField ($set = null);
+   *
+   * @param string $token
+   */
+  function setActive ($token);
```
Deviamos usar o método `activeField($active = null)` para alterar o estado do utilizador, o qual deve ser chamado depois de findByXXX().

claudio-silva commented on Interfaces/UserInterface.php:
```diff
@@ -168,4 +194,49 @@ function usernameField ($set = null);
+   * @param string $token
+   * @return UserInterface if the user was found
+   */
+  function getUserByRememberToken($token);
```
Isto parece-me ser uma dupliucação de findByRememberToken().

claudio-silva commented on Interfaces/UserInterface.php:
```diff
@@ -168,4 +194,49 @@ function usernameField ($set = null);
+   *
+   * @param string $email
+   */
+  function removeByEmail ($email);
```
Deveria ser mudado para `remove()`, já que o utilizador pode ser localizado de várias formas usando um dos métodos `findByXXX()`.

tiagomgelias committed:
Fix in some functions (name of function, parameters)

tiagomgelias committed:
Changes on function names

tiagomgelias committed:
FIX: CODE STYLE

claudio-silva commented on Interfaces/UserInterface.php:
```diff
@@ -62,6 +62,22 @@ function findById ($id);
+   * @param string $token
+   * @return bool True if the user was found.
+   */
+  function findByRememberToken ($token);
```
passa a findByToken()

tiagomgelias committed:
Mudança de nome de uma função e novo campo enabled

-> Mudança do nome da antiga função findByRememberToken para findByToken
-> Adicionado novo getter e setter para novo campo "enabled"

tiagomgelias committed:
Remoção dos métodos xxxField(), mudança de nome do getRecord e setRecord

setRecord() -> mergeFields()
getRecord() -> getFields()
